CMakeLists.txt

```diff
@@ -0,0 +1,423 @@
+set_or_default(NCCL_DIR NCCL_PATH)
+set_or_default(Thrust_DIR THRUST_PATH)
+set_or_default(CUDA_TOOLKIT_ROOT_DIR CUDA)
+set_or_default(CMAKE_CUDA_ARCHITECTURES GPU_ARCH ALL)
+
+option(Legion_SPY "Enable detailed logging for Legion Spy" OFF)
+option(Legion_USE_LLVM "Use LLVM JIT operations" OFF)
+option(Legion_USE_CUDA "Enable Legion support for the CUDA runtime" OFF)
+option(Legion_USE_HDF5 "Enable support for HDF5" OFF)
+option(Legion_USE_GASNet "Enable the distributed GASNet backend" OFF)
+option(Legion_USE_OpenMP "Use OpenMP" OFF)
+option(Legion_USE_Python "Use Python" OFF)
+
+if(Legion_USE_GASNet AND (NOT GASNet_ROOT_DIR))
+  option(Legion_EMBED_GASNet "Embed a custom GASNet build into Legion" ON)
+endif()
+
+set_or_default(GASNet_CONDUIT CONDUIT "udp")
+
+option(LEGATE_CORE_EXCLUDE_LEGION_FROM_ALL "Exclude Legion targets from legate.core's 'all' target" OFF)
+
+include(CMakeDependentOption)
+cmake_dependent_option(Legion_HIJACK_CUDART
+  "Allow Legion to hijack and rewrite application calls into the CUDA runtime"
+  ON
+  Legion_USE_CUDA;"DEFINED ENV{USE_CUDART_HIJACK} AND $ENV{USE_CUDART_HIJACK} STREQUAL 1"
+  OFF)
+
+if(Legion_USE_CUDA)
+  if(BUILD_SHARED_LIBS)
+    if(Legion_HIJACK_CUDART)
+      # Must statically link CUDA if HIJACK_CUDART is set
+      set(Legion_CUDA_DYNAMIC_LOAD OFF)
+    elseif(NOT DEFINED Legion_CUDA_DYNAMIC_LOAD)
+      # If HIJACK_CUDART isn't set and BUILD_SHARED_LIBS is true, default Legion_CUDA_DYNAMIC_LOAD to true
+      set(Legion_CUDA_DYNAMIC_LOAD ON)
+    endif()
+  elseif(NOT DEFINED Legion_CUDA_DYNAMIC_LOAD)
+    # If BUILD_SHARED_LIBS is false, default Legion_CUDA_DYNAMIC_LOAD to false also
+    set(Legion_CUDA_DYNAMIC_LOAD OFF)
+  endif()
+endif()
+
+if(Legion_USE_CUDA)
+  include(cmake/Modules/cuda_arch_helpers.cmake)
+  # Needs to run before `rapids_cuda_init_architectures`
+  set_cuda_arch_from_names()
+  # Needs to run before `project`
+  rapids_cuda_init_architectures(LEGATE_CORE)
+  project(LEGATE_CORE VERSION 22.05.00 LANGUAGES C CXX CUDA)
+else()
+  project(LEGATE_CORE VERSION 22.05.00 LANGUAGES C CXX)
+endif()
+
+# Write the version header
+rapids_cmake_write_version_file(include/legate/version_config.hpp)
+
+##############################################################################
+# - Build Type ---------------------------------------------------------------
+
+# Set a default build type if none was specified
+rapids_cmake_build_type(Release)
+
+# Needed to integrate with LLVM/clang tooling
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+##############################################################################
+# - Dependencies -------------------------------------------------------------
+
+# add third party dependencies using CPM
+rapids_cpm_init()
+
+if(Legion_USE_CUDA)
+  include(cmake/thirdparty/get_nccl.cmake)
+  include(cmake/thirdparty/get_thrust.cmake)
+endif()
+
+if(Legion_USE_Python)
+  find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
```

Comment:
should we always find Python3 for Legate (not only when `Legion_USE_Python`)?

Reply:
The C++ library can be built with or without Python support, so it does need to be an option. I'm overloading `Legion_USE_PYTHON` here because (presumably) building legate core's C++ Python support isn't useful unless legion/realm are also built with Python support. Though that may be an incorrect assumption?

Reply:
What is default for the `Legion_USE_Python`?

Reply:
In legion it's `OFF`, and in `install.py` it's hard-coded to `ON`.